src/gui_common/ModalManager.cs

```diff
@@ -98,17 +98,22 @@ public bool HideTopMostPopup()
         if (popup.Exclusive && !popup.ExclusiveAllowCloseOnEscape)
             return false;
 
-        // This is emitted before closing to allow window using components to differentiate between "cancel" and
-        // "any other reason for closing" in case some logic can be simplified by handling just those two situations.
-        if (popup is CustomWindow dialog)
-            dialog.EmitSignal(nameof(CustomWindow.Cancelled));
-
-        popup.Close();
-        popup.Notification(Control.NotificationModalClose);
+        HideModal(popup);
 
         return true;
     }
 
+    /// <summary>
+    ///   Attempt to clear all open modals.
+    /// </summary>
+    public void ClearModals()
```

Comment:
Does the HideModal method cause a modal to be removed from the stack? If so doesn't the foreach cause a crash by continuing to iterate? Maybe instead the loop should check if the stack is empty and if not attempt to close the top most one.

If this doesn't remove the modals in the stack, then I'm still of the opinion that the name of this method is wrong and should really be something like `ForceHideModals`.

Reply:
Nah it doesn't remove from the stack, happy to rename it.

Reply:
Alright, then the foreach doesn't crash. And yeah I'd really like it to be renamed if the method doesn't even "clear" the modals from the data structure.